controllers/barbicanapi_controller.go

```diff
@@ -326,6 +327,45 @@ func (r *BarbicanAPIReconciler) generateServiceConfigs(
 		"EnableSecureRBAC": instance.Spec.EnableSecureRBAC,
 	}
 
+	// Checking if there's an HSM.
+	pkcs11 := instance.Spec.PKCS11
+	if pkcs11.HSMEnabled {
+		hsmLoginSecret, _, err := secret.GetSecret(ctx, h, pkcs11.HSMLogin, instance.Namespace)
+		if err != nil {
+			return err
+		}
+		hsmCertificatesSecret, _, err := secret.GetSecret(ctx, h, maps.Keys(pkcs11.HSMCertificates)[0], instance.Namespace)
```

Comment:
You do not need this secret here.  This is mounted by the volumes/ volume mounts